appshell/appshell_extensions_gtk.cpp

```diff
@@ -758,6 +762,92 @@ static void Callback(GtkMenuItem* menuItem, gpointer userData) {
     }
 }
 
+
+ExtensionString GetDisplayKeyString(const ExtensionString& keyStr)
+{
+    ExtensionString result = keyStr;
+
+    // We get a keyStr that looks like "Ctrl-O", which is the format we
+    // need for the accelerator table. For displaying in the menu, though,
+    // we have to change it to "Ctrl+O". Careful not to change the final
+    // character, though, so "Ctrl--" ends up as "Ctrl+-".
+    if (result.size() > 0) {
+        replace(result.begin(), result.end() - 1, '-', '+');
+    }
+    return result;
+}
+
+
+int32 ParseShortcut(CefRefPtr<CefBrowser> browser, GtkWidget* entry, ExtensionString& key, GdkModifierType* modifier, guint* keyVal, ExtensionString& commandId) {
+    if (key.size()) {
+        GtkAccelGroup *accel_group = NULL;
+        accel_group = gtk_accel_group_new();
+
+        std::vector<std::string> tokens;
+        std::istringstream f(key.c_str());
+        std::string s;
+        while (getline(f, s, '-')) {
+            tokens.push_back(s);
```

Comment:
Actually I have used tokens in creating shortcut of type <Ctrl>A and also in the fallback, so I won't be able to have fallback if there is no tokens vector.